fireant/dataset/data_blending.py

```diff
@@ -0,0 +1,108 @@
+from fireant import (
+    Field,
+)
+from fireant.dataset.klass import (
+    DataSet,
+)
+from fireant.queries import (
+    DimensionChoicesQueryBuilder,
+    DimensionLatestQueryBuilder,
+)
+from fireant.queries.builder.dataset_blender_query_builder import DataSetBlenderQueryBuilder
+from fireant.utils import immutable
+
+
+class DataSetBlender(object):
+    """
+    WRITEME
+    """
+
+    def __init__(self, primary_dataset):
+        """
+        Constructor for a data blended slicer.  Contains all the fields to initialize the slicer.
+
+        :param primary_dataset: (Required)
+            The primary dataset, which table will be used as part of the FROM expression.
+        """
+        self.primary_dataset = primary_dataset
+        self.table = primary_dataset.table
+        self.secondary_datasets = []
+        self.fields = DataSet.Fields()
+
+        self.field_mapping = {}
+
+        primary_dataset_field_mapping = {}
+
+        for primary_field in self.primary_dataset.fields:
+            primary_dataset_field_mapping[primary_field] = primary_field
+
+        self.field_mapping[primary_dataset.table._table_name] = primary_dataset_field_mapping
+
+        for primary_field in self.primary_dataset.fields:
+            if primary_field.alias not in self.fields:
+                self.fields.append(primary_field)
+
+        # add query builder entry points
+        self.query = DataSetBlenderQueryBuilder(self)
+        self.latest = DimensionLatestQueryBuilder(self.primary_dataset)
+
+    def __eq__(self, other):
+        return isinstance(other, DataSetBlender) \
+               and self.fields == other.fields
+
+    def __repr__(self):
+        return 'BlendedSlicer(fields=[{}])' \
+            .format(','.join([repr(f)
+                              for f in self.fields]))
+
+    def __hash__(self):
+        return hash((
+            self.table, self.primary_dataset.database, tuple(self.secondary_datasets), self.joins, self.fields,
+        ))
+
+    @immutable
+    def join(self, secondary_dataset, field_mapping=None):
+        """
+        Adds a join when building a slicer query. The join type is left outer by default.
+
+        :param secondary_dataset: (Required)
+            The secondary dataset, which table will be used as part of the one or more JOIN expressions.
+        :param field_mapping: (Optional)
```

Comment:
would be really nice to provide some example of what this should look like in the docstring

Reply:
Examples were added.